### ISSUE TITLE
Support additional (cloud-init) network config in the xenorchestra_vm resource 

### ISSUE THREAD

**Issue by marlluslustosa** [terra-farm/terraform-provider-xenorchestra#34]
I usually use Xen Orchestra to load a cloud-config specific to a VM by also defining the network parameters in an additional box.

[![Screenshot-20200724-193418.png](https://raw.githubusercontent.com/marlluslustosa/images_forum/master/terraform-xoa/Screenshot_20200724_193418.png)

So when I used terraform, I saw that it is only supported to create the "user-data" file and not also the "network-config". So I can't create the network settings and set it on the newly created VM, as it happens without problems in Xen Orchestra. If I put the network settings together with xenorchestra_cloud_config it just doesn't work.

According to [cloud-init documentation](https://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html#datasource-nocloud):

> Network configuration can also be provided to cloud-init in either Networking Config Version 1 or Networking Config Version 2 by providing that yaml formatted data in a **file named network-config**. If found, this file will override a network-interfaces file.

So it would be interesting to have a way to also create a "network-config" resource to also create this file in the main directory.

Any idea?

**Comment by brd** [comment]
This requires defining something very similar to `cloud_config` on the xenorchestra_vm resource, perhaps called `network_config`?

**Comment by brd** [comment]
@ddelnano Can you take a look at this?  I can't find any docs on the xen orchestra API..

**Comment by ddelnano** [comment]
Sorry for the delay on this. I believe this network config is just another parameter to the `vm.create` rpc call.

```
ddelnano@ddelnano-desktop:~/go/src/github.com/ddelnano/terraform-provider-xenorchestra$ xo-cli --list-comm
ands | grep vm.create
vm.create .... [networkConfig=<string>] ....
```

This should be easy to support but I'm not sure that this should be its own resource though. The XO api doesn't have a "network config" object. While the network config is ultimately added to the cloud-init drive attached the the VM, the cloud config object in the api does not support anything aside from a template. The API does have the option to create a cloud init config drive with network config but I think this just creates a VDI. That would then mean that the provider would need to handle attaching this VDI to the VM which imo feels more difficult to use.

```
vm.createCloudInitConfigDrive vm=<string> sr=<string> config=<string> [networkConfig=<string>]
```

So I think the best path forward is to add an optional `network_config` attribute to the `xenorchestra_vm` resource and to pass it directly into the `vm.create` rpc call.

**Comment by brd** [comment]
Yeah, it doesn't need its own resource, just a way to apply a template to the `xenorchestra_vm` resource.

**Comment by ddelnano** [comment]
This fix will be released in v0.9.1

**Comment by brd** [comment]
Works like a charm, thanks @ddelnano !
